libs/injection/mlpe/injection/injection.py

```diff
@@ -1,81 +1,94 @@
+def generate_time_domain_sine_gaussian(
+    qualities: np.ndarray,
+    frequencies: np.ndarray,
+    hrss: np.ndarray,
+    phases: np.ndarray,
+    eccentricities: np.ndarray,
     sample_rate: float,
-    waveform_duration: float,
-    waveform: Callable,
-    waveform_arguments: Optional[Dict] = None,
-    parameter_conversion: Optional[Callable] = None,
+    duration: float,
 ):
+
+    """
+    Generate time domain sine-Gaussian waveforms using lalinference. See
+    https://git.ligo.org/lscsoft/lalsuite/-/blob/master/lalinference/lib/LALInferenceBurstRoutines.c#L381
+    for more details on parameter definitions.
     """
-    Generate raw frequency domain gravitational-wave signals
-    pre-interferometer projection.
 
-    Args:
-        sample_params:
-            Dictionary of GW parameters where key is the parameter name
-            and value is a list of the parameters
-        sample_rate:
-            Rate at which to sample time series
-        waveform_duration:
-            Duration of waveform
-        waveform:
-            Callable whose first argument is a frequency array
-            and returns returns frequency domain
-            hplus and hcross polarizations at the specified frequencies
-            for a given set of parameters that are the following arguments
-        waveform_arguments:
-            A dictionary of fixed keyword arguments to pass to
-            frequency_domain_source_model via bilby waveform generator
-        parameter_conversion:
-            Callable to convert from sampled parameters
-            to parameters of the waveform generator.
-            Default value is the identity,
-            i.e. it leaves the parameters unaffected.
-            Typically used when generating
-            CBC waveforms via lal_binary_black_hole
-
-    Returns:
-        An (n_samples, 2, waveform_size) array, containing both polarizations
-        for each of the desired number of samples.
-        The waveforms are shifted such that
-        the coalescence time lies at the center of the sample
+    delta_t = 1 / sample_rate
+    samples = zip(qualities, frequencies, hrss, eccentricities, phases)
+    cross, plus = [], []
+    for sample in samples:
+
+        # calculate waveform with lalsimulation
+        hplus, hcross = BurstSineGaussian(*sample, delta_t=delta_t)
+        hplus = hplus.data.data
+        hcross = hcross.data.data
+
+        # pad with zeros to desired waveform duration
+        waveform_size = int(sample_rate * duration)
+
+        total_pad = waveform_size - len(hplus)
+        pad_left = total_pad // 2
+        pad_right = total_pad - pad_left
+
+        hplus = np.pad(hplus, (pad_left, pad_right), mode="constant")
+        hcross = np.pad(hcross, (pad_left, pad_right), mode="constant")
+
+        cross.append(hcross)
+        plus.append(hplus)
+
+    cross = np.stack(cross)
+    plus = np.stack(plus)
+    return cross, plus
+
+
+def sine_gaussian_frequency(
```

Comment:
rename to `generate_frequency_domain_sine_gaussian` similar to its time domain counterpart.

Reply:
Since it's only really meant to be used with bilby, will do `bilby_frequency_domain_sine_gaussian` to clarify that